app/src/main/java/org/mozilla/focus/fragment/HomeFragment.kt

```diff
@@ -55,7 +59,12 @@ class HomeFragment : Fragment() {
     }
 
     private fun initTiles() = with (tileContainer) {
-        adapter = HomeTileAdapter(onUrlEnteredListener)
+        var homeTiles = mutableListOf<JSONObject>()
+        val jsonArray = HomeTileUtils.loadDefaultTiles(context).getJSONArray("default_tiles")
+        for (i in 0..(jsonArray.length() - 1)) {
```

Comment:
A forEach here would be better - maybe @mcomella has an even better kotlin suggestion!

Reply:
You could create an extension function for JSONArray like, `JSONArray.forEach` so no one else is tortured by this in the future, but otherwise I think this is the only way to iterate with this API.

For completeness, another suggestion would be using JSON APIs that support iterators (like Klaxon) but we've already discussed this and you've already made that decision. :)